### PR TITLE
Allow controller property to be set in key patches

### PR DESCRIPTION
**Pull request by csuwildcat**

### CODE REVIEW

File: docs/spec/patches.md [hunk 1]
```diff
@@ -37,10 +37,11 @@ The `add-public-keys` _Patch Action_ describes the addition of cryptographic key
...
-    4. The object ****MAY**** include a `purposes` property, and if included, its value ****MUST**** be an array of one or more strings. The value for each string ****SHOULD**** represent a verification relationship defined by [DID Core](https://w3c.github.io/did-core) or the [DID Specification Registries](https://w3c.github.io/did-spec-registries). If the value is not of the correct type or contains any string not listed below (or defined by the implementor), the entire _Patch Action_ ****MUST**** be discarded, without any of it being used to modify the DID's state.
+    3. The object ****MAY**** include a `controller` property, and its value ****MUST**** be a DID URI string of no more than one hundred (100) Base64URL encoded characters. If the `controller` property is absent, the implementation ****must**** set the corresponding property in the resolved DID Document with a value that equates to the DID Document controller's id. If the value is not of the correct type or exceeds the specified length, the entire _Patch Action_ ****MUST**** be discarded, without any of the patch being used to modify the DID's state.
```
Comment (troyronda) [inline docs/spec/patches.md:42]: Does 100 Base64URL encoded characters cover most DID URIs?

Comment (troyronda) [reply docs/spec/patches.md:42]: > Base64URL encoded characters

I would assume a _valid_ DID string should just be put directly into the property?

Comment (csuwildcat) [reply docs/spec/patches.md:42]: Probably will change to implementation defined limit

File: docs/spec/patches.md [hunk 2]
```diff
@@ -37,10 +37,11 @@ The `add-public-keys` _Patch Action_ describes the addition of cryptographic key
...
+    4. The object ****MUST**** include either a `publicKeyJwk` or a `publicKeyMultibase` property. Implementors ****MAY**** choose to only define `publicKeyJwk`. These key representations are described in the JWK and Multibase subsections.
```
[the remark this answers is not in the document]
Comment (csuwildcat) [reply docs/spec/patches.md:43]: depends, but perhaps the best course is to specify this as 'implementation defined'

**Pull request merged**